app/models/user_answers/your_home.py

```diff
@@ -13,3 +18,18 @@ class YourHomeAnswers(BaseModel):
     people_in_house: conint(ge=1, le=6)
     postcode: constr(strip_whitespace=True, pattern=r"^\d{4}$")
     disconnect_gas: bool
+
+    @model_validator(mode="after")
+    def check_postcode(cls, model):
+        """
+        Validate that the postcode is recognized (a key from the
+        climate zone lookup)
+
+        Raises:
+            ValueError: If the postcode is not in the known list.
+        """
+        if f"{model.postcode}" not in known_postcodes:
+            raise ValueError(
+                f"Invalid postcode '{model.postcode}'. Please provide a valid postcode."
+            )
+        return model
```

Comment:
![image](https://github.com/user-attachments/assets/3fa8258c-76fa-478a-875a-31e6042c456a)

not a biggie other than the typo "Post[space]Code". Could be a Deviate thing.

and for the excluded postcodes, the error message could be slightly changed?

![image](https://github.com/user-attachments/assets/0a536fcb-4521-4363-b439-1fb0d0a73321)

Reply:
Thanks Achini, have passed on to the web developers.